### PR TITLE
loading state for routes

### PR DESCRIPTION
**Pull request by akshay1502**
Didn't added error state for tasks, since we are handling it in the model.

### CODE REVIEW

File: app/routes/tasks.js [hunk 1]
```diff
@@ -24,4 +25,9 @@ export default class TasksRoute extends Route {
       );
     }
   };
+
+  @action
+  loading() {
+    return true;
+  }
```
Comment (swarajpure) [inline app/routes/tasks.js:32]: When are we setting this to false?

Comment (akshay1502) [reply app/routes/tasks.js:32]: https://guides.emberjs.com/release/routing/loading-and-error-substates/#toc_the-loading-event
by doing this ember searches for `tasks-loading.hbs`.

Comment (swarajpure) [reply app/routes/tasks.js:32]: Well, they're setting it false after the promise resolves. Are we doing that?

Comment (akshay1502) [reply app/routes/tasks.js]: Well, I found out we don't even have to do that. Just add `routename-loading` file.

**Comment by akshay1502** [comment]
https://user-images.githubusercontent.com/55057874/147391013-b24378e7-173a-42b1-83a7-ad530096b792.mp4

File: app/routes/tasks.js (hunk 1 above)
Comment (swarajpure) [reply app/routes/tasks.js]: but if we have 15 routes tomorrow, we'll need 15 `*-loading.hbs* files, right? 🤔

**Comment by akshay1502** [comment]
https://user-images.githubusercontent.com/55057874/147734344-3b6e0e08-2096-4481-98b7-bc433537f73b.mp4

**Comment by akshay1502** [comment]
https://user-images.githubusercontent.com/55057874/147804153-8353ede1-62e9-495f-b0a6-d8a602ae0e86.mp4

File: app/styles/app.css [hunk 2]
```diff
@@ -46,3 +46,9 @@ body {
   width: 100%;
   padding: 16px;
 }
+
+.loading {
+  text-align: center;
+  margin-top: 5%;
+  font-size: 24px;
+}
```
Comment (swarajpure) [inline app/styles/app.css:54]: Can we have the loader in center of the screen?
Margin-top 5% really a bit off, creates a lot of empty space on rest of the page. After this change, we'll be good with merging the PR 🥳

**Comment by akshay1502** [comment]
https://user-images.githubusercontent.com/55057874/147848856-c7155ac8-5347-4a27-82cd-6602a25c8f63.mp4

**Review by swarajpure — APPROVED** [review]
Great job in there! 🔥 
Thank you so much for caring about the UX ⚡

**Comment by akshay1502** [comment]
@swarajpure thanks for the guidance and reviews too.

**Pull request merged**